mycpp/gc_str.cc

```diff
@@ -482,3 +488,96 @@ List<Str*>* Str::split(Str* sep) {
 
   return result;
 }
+
+// TODO: We should reduce the number of allocations and copies done here.
+static inline Str* _StrFromFormat(const char* fmt, va_list args) {
+  int n = strlen(fmt);
+  auto beg = std::cregex_iterator(fmt, fmt + n, gStrFmtRegex);
+  auto end = std::cregex_iterator();
+  mylib::BufWriter* w = Alloc<mylib::BufWriter>();
+  for (std::cregex_iterator match = beg; match != end; ++match) {
+    auto& lit = (*match)[1];
+    assert(lit.matched);
+    w->write(StrFromC(lit.str().c_str(), lit.str().size()));
+
+    bool zero_pad = false;
+    int width = 0;
+    auto& width_m = (*match)[2];
+    if (width_m.matched && !width_m.str().empty()) {
+      if (width_m.str()[0] == '0') {
+        zero_pad = true;
+        assert(width_m.str().size() > 1);
+        width = atoi(&width_m.str()[1]);
```

Comment:
atoi() is generally buggy, and we have some functions that wrap strtol() in gc_builtins

But I guess for this case it should be OK since the regex in validates it?  That is worth a command

Reply:
Ah great. We might as well use the helpers